backend/src/database/migrations/20230331185335_create_db_lock_table.js

```diff
@@ -0,0 +1,13 @@
+exports.up = function (knex) {
+  return knex.schema.withSchema("data").createTable("db_lock", function (table) {
+    table.increments().primary(); //id column
+    table.string("locked_table");
+    table.string("locked_by");
+    table.integer("locked_row_id").unsigned();
+    table.date("locked_date");
```

Comment:
We could make the primary key a composite of `locked_table`, `locked_by`, `locked_row_id` and get rid of the `id` column or if you still want the `id` column we could just add a UNIQUE constraint on those columns so that it's not possible to add duplicates.

Reply:
This is a tricky one we need to discuss further.  At this time I don't want to change how it is.